gtk4/src/dialog.rs

```diff
@@ -49,9 +67,21 @@ impl<O: IsA<Dialog>> DialogExtManual for O {
     fn get_response_for_widget<P: IsA<Widget>>(&self, widget: &P) -> ResponseType {
         unsafe {
             from_glib(ffi::gtk_dialog_get_response_for_widget(
-                self.as_ref().to_glib_none().0,
+                AsRef::<Dialog>::as_ref(self).to_glib_none().0,
                 widget.as_ref().to_glib_none().0,
             ))
         }
     }
+
+    fn run<F: FnOnce(&Self, ResponseType) + 'static>(&self, f: F) {
+        let response_handler = Arc::new(RefCell::new(None));
```

Comment:
Just a `RefCell<Option<_>>` is enough

Reply:
Oh, no needs to be an `Rc` around it too (but not `Arc`).

But you need a `RefCell<Option<_>>` for the closure so you can call it exactly once (it's `FnOnce`!)